modules/tabs/test/spec.coffee

```diff
@@ -1,2 +1,212 @@
 describe 'hx-tabs tests', ->
-  # ...
\ No newline at end of file
+  getHeaders = (root) ->
+    root.selectAll '.hx-tab'
+      .map (header) -> { header }
+
+  setupTabsUsingStructuredObject = ->
+    items = [{
+      title: 'title1',
+      content: {
+        name: 'Bob',
+        town: 'Wheathampstead'
+      }
+    }, {
+      title: 'title2',
+      content: {
+        name: 'Kate',
+        town: 'North Mymms'
+      },
+    }, {
+      title: 'title3',
+      context: 'negative',
+      content: {
+        name: 'Ganesh',
+        town: 'Digswell'
+      }
+    }, {
+      title: 'title4',
+      context: 'action',
+      content: {
+        name: 'Lazlo',
+        town: 'Essendon'
+      }
+    }]
+
+    titles =
+      title1: 'Example1'
+      title2: 'Example2'
+      title3: 'Example3'
+      title4: 'Example4'
+
+    titleRenderer = (node, data) -> hx.select(node).text(titles[data])
+
+    contentRenderer = chai.spy (node, item) ->
+      { name, town } = item
+      hx.select node
+        .add hx.detached('span').text name
+        .add hx.detached('span').text town
+
+    hx.tabs { titleRenderer, contentRenderer, items }
+
+  setupTabs = ->
+
+    ctx = [undefined, undefined, 'negative', 'action']
+    createTabHeaderAndBody = (i) ->
+      identifier = "tab-content-#{i}"
+      header = hx.detached 'div'
+        .class 'hx-tab'
+        .attr 'id', "tab-#{i}"
+        .attr 'data-content', identifier
+      hx.palette.context header.node(), ctx[i]
+      body = hx.detached 'div'
+        .attr 'id', identifier
+        .class 'hx-tab-content'
+      { header, body }
+
+    numberOfTabs = ctx.length
+
+    headerAndBody = hx.range numberOfTabs
+      .map createTabHeaderAndBody
+
+    tabsContent = hx.detached 'div'
+      .class 'hx-tabs-content'
+      .add headerAndBody.map ({ body }) -> body
+
+    rootSel = hx.detached 'div'
+      .add  headerAndBody.map ({ header }) -> header
+      .add tabsContent
+
+    tabs = new hx.Tabs rootSel.node()
+
+    { headerAndBody, rootSel, tabs }
+
+  verifyVisible = (headerAndBody, i) ->
+    expectedVisible = headerAndBody.map (_, index)  -> index is i
+
+    actualVisible = headerAndBody.map ({ header }) ->
+      header.classed 'hx-tab-active'
+
+    actualVisible.should.eql expectedVisible
+
+  testChangeEventFired = (cause, tabToSelect, done) ->
```

Comment:
This name is a bit weird - It's not actually testing anything on it's own, you have to call something else to run the test - could this be renamed to `setupChangeEventTest` or something like that?